internal/datacoord/services.go

```diff
@@ -1545,16 +1568,13 @@ func (s *Server) BroadcastAlteredCollection(ctx context.Context, req *datapb.Alt
 
 	// cache miss and update cache
 	if clonedColl == nil {
-		collInfo := &collectionInfo{
-			ID:             req.GetCollectionID(),
-			Schema:         req.GetSchema(),
-			Partitions:     req.GetPartitionIDs(),
-			StartPositions: req.GetStartPositions(),
-			Properties:     properties,
-			DatabaseID:     req.GetDbID(),
+		err := s.loadCollectionFromRootCoord(ctx, req.GetCollectionID())
```

Comment:
any reason for this modification?

Reply:
req has no vchannel info, have to get it from rootcoord

Reply:
Will it be a better choice to modify the proto of req?